hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassCluster.java

```diff
@@ -164,6 +170,8 @@ private void readProperties() {
         keyspace = AlertProperties.getProperty(ALERTS_CASSANDRA_KEYSPACE, "hawkular_alerts");
         cassandraUseSSL = Boolean.parseBoolean(AlertProperties.getProperty(ALERTS_CASSANDRA_USESSL,
                 ALERTS_CASSANDRA_USESSL_ENV, "false"));
+        maxQueue = Integer.parseInt(AlertProperties.getProperty(ALERTS_CASSANDRA_MAX_QUEUE,
+                ALERTS_CASSANDRA_MAX_QUEUE_ENV, "9182"));
```

Comment:
This setting is related to your previous comment:

> I am not against the driver-based batching, I think it could help our response times. I think even with the batching introduced here we may still run into the thread issues that have shown up with the change to the 3.0.5 driver. To avoid that we may still need to limit the number of futures we submit concurrently. So we may still need to chunk large collections. For example, if storing 1000 events, we may need to chunk by 100 and then batch those into 10s.

I see your point here, and I guess that at some point we should not discard that. But in general, this setting is doing basically the same at driver level. It queues all the queries that driver can not process.

If we batch/cut the max number of parallel async queries will help, but I guess that driver is doing the same.
Point is that controlling this for the application on an efficient way could be costly, as we need to store another singleton state with the max queries per all request (not only per a single request).
And also, reading all best practices with C* for large writtings the recommendation is basically what we are doing, batch in unlogged manner for same partition, and make async queries in general.